pyne/endl.py

```diff
@@ -0,0 +1,239 @@
+#!/usr/bin/env python
+
+"""Module for parsing and manipulating data from ENDL evaluations.
+
+For the moment, classes and functions in this module only implement the
+specifications relevant to the EEDL (Evaluated Electron Data Library) and the
+EPDL (Evaluated Photon Data Library). The formats are described by the
+following documents:
+
+"ENDL type formats for the Livermore Evaluated Photon Data Library, EPDL"
+https://www.oecd-nea.org/dbdata/data/manual-endf/nds_eval_epdl.pdf
+
+"ENDL type formats for the Livermore Evaluated Electron Data Library, EEDL"
+https://www.oecd-nea.org/dbdata/data/manual-endf/nds_eval_eedl.pdf
+
+For more information, contact Davide Mancusi <davide.mancusi@cea.fr>.
+"""
+from __future__ import print_function, division, unicode_literals
+
+from warnings import warn
+from pyne.utils import QAWarning
+
+import pyne.rxdata as rx
+import pyne.utils as utils
+from pyne import nucname
+
+import numpy as np
+import re
+
+warn(__name__ + ' is not yet QA compliant.', QAWarning)
+
+END_OF_TABLE_RE = re.compile(' {71}1')
+
+
+class Library(rx.RxLib):
+    """A class for a file which contains multiple ENDL tables."""
+    def __init__(self, fh):
+        self.mts = {}
+        self.structure = {}
+        self.mat_dict = {}
+        self.intdict = {
+                0: self._linlin,
+                2: self._linlin,
+                3: self._loglin,
+                4: self._linlog,
+                5: self._loglog,
+                }
+        self.fh = fh
+        # read headers for all tables
+        self._read_headers()
+
+#    def load(self):
+#        """load()
+#        Read the ENDL file into a NumPy array.
+#
+#        Returns
+#        -------
+#        data : np.array, 1d, float64
+#            Returns a 1d float64 NumPy array.
+#        """
+#        opened_here = False
+#        if isinstance(self.fh, basestring):
+#            fh = open(self.fh, 'r')
+#            opened_here = True
+#        else:
+#            fh = self.fh
+#        fh.readline()
+#        data = fromendf_tok(fh.read())
+#        fh.seek(0)
+#        if opened_here:
+#            fh.close()
+#        return data
+
+    def _read_headers(self):
+        opened_here = False
+        if isinstance(self.fh, basestring):
```

Comment:
basestring is not Python 3 compatible. In other modules in pyne, this variable is defined as str in the event of Python 3.